### ISSUE TITLE
Angular elements disabled with v1.9.0

### ISSUE THREAD

**Issue by gautamdsheth** [pnp/generator-spfx#201]
#### General Information
* [ ] Usage
* [x] Development
* [ ] Documentation
* [x] Feature Request

#### Specific generator
* [ ] Addons
* [ ] HandlebarsJS

#### What Version you are running?
* NodeJS:  10.16.3
* NPM: 6.5.0
* @pnp/spfx: 1.9.0 
* Angular CLI : 8.3.2

Using the 1.9.0 version of the generator, the Angular elements option is disabled.

Steps to repro:

Install the latest generator and the Angular CLI

`npm install -g @pnp/generator-spfx @angular/cli`

Looks like when installing the generator in a fresh machine, the Angular elements option is disabled.

The reason behind this is that the latest Angular CLI downloaded version is 8.3.2.
The generator only checks if the version 6 or 7 is installed. 
Need to add additional check for version 8. 

Followed the instructions from [here](https://pnp.github.io/generator-spfx/howtos/angularelements/).

Temporary workaround while using generator 1.9 is to use Angular CLI v6 or v7 as below:

`npm install -g @pnp/generator-spfx @angular/cli@6.2.9`

or 

`npm install -g @pnp/generator-spfx @angular/cli@7.3.9`

**Comment by StfBauer** [comment]
During testing Angular 6/7/8 we recognised some difficulties bundle and loading the output of Angular 8. For now, Angular 8 is not supported until these issues are resolved.

If you find a workaround or a solution on how to bundle Angular 8, we are looking actively for help on this.

That currently Angular 6/7 are only supported was also mentioned in the release note.

@gautamdsheth Can you maybe help us with this?

Thank you.

**Comment by gautamdsheth** [comment]
Gotcha, my bad. Sorry about the PR :(

Will absolutely also try to find a way to make Angular v8 work with SPFx.

Thanks a lot for the quick reply, love the stuff that you and the team does ! Really appreciate it :)

**Comment by StfBauer** [comment]
I order to test the generator the only thing you need to do is to disable the checks, then you should be able to use Angular 8. 

From my tests I figured out:
- Bundle Script is not working anymore because NG8 generates different files
- When you rewrite the bundle script the component for some reason doesn't load

If you need anything let me know.

**Comment by gautamdsheth** [comment]
hey @StfBauer , yes looks like in NG8 they are generating different files. If we include those files in the SPFx webpart, i was able to get it working. 

You can check the working demo at my repo - https://github.com/gautamdsheth/SPFxNG8 

I updated the SPFx webpart code based on this link - https://angular.io/guide/deployment#differential-builds

Basically added the 4 additional files in the SPFx webpart in the order specified and then it started working. Kinda reverse engineering :( 

Not sure how we can handle this via the generator considering that we also support v6 and 7.

**Comment by StfBauer** [comment]
Hey, @gautamdsheth so you added the following files directly to the web part, right?

```javascript
import 'testsol/dist/testsol/runtime.js';
import 'testsol/dist/testsol/polyfills.js';
import 'testsol/dist/testsol/styles.js';
import 'testsol/dist/testsol/vendor.js';
import 'testsol/dist/testsol/main.js';
```

How have you built that only those files generate? Normally it should build something like theses files:

```html
<script src="runtime-es2015.js" type="module"></script>
  <script src="runtime-es5.js" nomodule></script>
  <script src="polyfills-es2015.js" type="module"></script>
  <script src="polyfills-es5.js" nomodule></script>
  <script src="styles-es2015.js" type="module"></script>
  <script src="styles-es5.js" nomodule></script>
  <script src="vendor-es2015.js" type="module"></script>
  <script src="vendor-es5.js" nomodule></script>
  <script src="main-es2015.js" type="module"></script>
  <script src="main-es5.js" nomodule></script>
```

In 6 and 7, we bundle all the required files together. For some reason, this doesn't work anymore.
Integration into the generator is not that big of a problem when I know the steps to take. I can implement a decision based on the CLI version installed.

In the Angular Project there is a npm script that says bundle that combines all required files.

**Comment by gautamdsheth** [comment]
ok sorry again :( . Please ignore the above comment.

I found the culprit for this issue in NG8. I had disabled the checks in the generator and created a fresh project in SPFx

Turns out when we have NG8 project created using the latest Angular CLI (v8), the `tsconfig.json` has the ECMA script target version is set to `es2015`. Once i change it to `es5`, everything works perfectly.

So, the only change required is now in the Angular project's `tsconfig.json` file.

By default, it is generated as :

```typescript
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "module": "esnext",
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es2015",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2018",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "fullTemplateTypeCheck": true,
    "strictInjectionParameters": true
  }
}

```

we need to change it to :

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "module": "esnext",
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2018",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "fullTemplateTypeCheck": true,
    "strictInjectionParameters": true
  }
}

```

To test it, simply create a new Angular elements projects with Angular CLI v8.
Try to execute gulp serve or any SPFx gulp commands, they will error out.
Now, modify the `tsconfig.json` file as mentioned above.
Execute the gulp commands or npm bundle commands, it will work smoothly.

**Comment by StfBauer** [comment]
So everything else then is back to normal like in NG6/7 right? The only issues is still when someone likes to use es2015.

**Comment by gautamdsheth** [comment]
Yes, only change is in the `tsconfig.json` of the Angular project.  Everything else is normal like NG 6/7. Yes, will be still an issue if someone likes to use es2015. 

So, i guess we need to write a post-angular project creation script to modify the `tsconfig.json`  file via the generator which will then handle the issue.

**Comment by StfBauer** [comment]
Thank you @gautamdsheth for you support it should now be fixed in 1.9.1 of pnp/spfx.

**Comment by gautamdsheth** [comment]
Hey @StfBauer , great stuff. Tested it today, works perfectly fine. Thanks a lot !
